SU2_CFD/include/numerics/NEMO/NEMO_diffusion.hpp

```diff
@@ -0,0 +1,258 @@
+/*!
+ * \file NEMO_slau.hpp
+ * \brief Declaration of numerics classes for the NEMO family of schemes,
+ *        including SLAU. The implementation is in NEMO.cpp.
+ * \author F. Palacios, T. Economon
+ * \version 7.0.3 "Blackbird"
+ *
+ * SU2 Project Website: https://su2code.github.io
+ *
+ * The SU2 Project is maintained by the SU2 Foundation
+ * (http://su2foundation.org)
+ *
+ * Copyright 2012-2020, SU2 Contributors (cf. AUTHORS.md)
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "../CNumerics.hpp"
+#include "../../variables/CNEMONSVariable.hpp"
+
+/*!
+ * \class CAvgGrad_NEMO
+ * \brief Class for computing viscous term using the average of gradients.
+ * \ingroup ViscDiscr
+ * \author S. Copeland, W. Maier
+ * \version 6.2.0 "falcon"
+ */
+class CAvgGrad_NEMO : public CNumerics {
+private:
+  unsigned short iDim, iVar, nPrimVar, nPrimVarGrad;		/*!< \brief Iterators in dimension an variable. */
+  su2double *Mean_PrimVar,					/*!< \brief Mean primitive variables. */
+  *Mean_U,
+  **Mean_GU,
+  *Mean_dTdU,
+  *Mean_dTvedU,
+  *Mean_dPdU,
+  *Mean_Eve,
+  *Mean_Cvve,
+  *PrimVar_i, *PrimVar_j,				/*!< \brief Primitives variables at point i and 1. */
+  **Mean_GradPrimVar,						/*!< \brief Mean value of the gradient. */
+  *Mean_Diffusion_Coeff, /*!< \brief Mean value of the species diffusion coefficient. */
+  Mean_Laminar_Viscosity, /*!< \brief Mean value of the viscosity. */
+  Mean_Thermal_Conductivity, /*!< \brief Mean value of the thermal conductivity. */
+  Mean_Thermal_Conductivity_ve, /*!< \brief Mean value of the vib-el. thermal conductivity. */
+
+  *ProjFlux,	/*!< \brief Projection of the viscous fluxes. */
+  dist_ij;						/*!< \brief Length of the edge and face. */
+  bool implicit; /*!< \brief Implicit calculus. */
+  CNEMOEulerVariable *variable;
```

Comment:
This is unusual, can you add some explanation?

Reply:
I think this comes into an issue where the numerics portion of the code currently require functions implemented in the variable class.  Work has been to fix this in a move to mimic the fluid model class.